From 2364c4b5647414d20f18acca9e409ae8a2d0925e Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Mon, 19 Feb 2024 14:45:35 +0100
Subject: [PATCH 0797/1695] dts: stm32: enable AXI address remapping for HPDMA
 on stm32mp21

Enable HPDMA AXI address remapping on stm32mp211.dtsi, so HPDMA can
access up to 4GB of DDR on its AXI port.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 08fd6a816b76dd4087c22d7ce82c034b363f20c9 dts: stm32: enable AXI address remapping for HPDMA on stm32mp21]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: If39d43ff8da1829fd0d99a0d89ad11031ab434a4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/372999
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index a6e02e5c3..5d5ce4be6 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -163,6 +163,7 @@
 			clocks = <&rcc CK_BUS_HPDMA1>;
 			resets = <&rcc HPDMA1_R>;
 			#dma-cells = <4>;
+			st,syscfg-arcr = <&syscfg 0x2050 0x1>;
 		};
 
 		hpdma2: dma-controller@40410000 {
@@ -171,6 +172,7 @@
 			clocks = <&rcc CK_BUS_HPDMA2>;
 			resets = <&rcc HPDMA2_R>;
 			#dma-cells = <4>;
+			st,syscfg-arcr = <&syscfg 0x2050 0x2>;
 		};
 
 		hpdma3: dma-controller@40420000 {
@@ -179,6 +181,7 @@
 			clocks = <&rcc CK_BUS_HPDMA3>;
 			resets = <&rcc HPDMA3_R>;
 			#dma-cells = <4>;
+			st,syscfg-arcr = <&syscfg 0x2050 0x4>;
 		};
 
 		ipcc1: mailbox@40490000 {
-- 
2.47.2

